File: C/24BCSH93/3rd Semester/OS/assignment6/question10.c
/*  Q10. Write a program that accepts an integer n from the user and creates only that many child processes. Each child process prints its own PID and
         then exits. The parent waits for all children before terminating.
*/

#include <stdio.h>
#include <unistd.h>
#include <wait.h>

int main() {
    int n, i;
    printf("Enter a number: ");
    scanf("%d", &n);

    for (i = 0; i < n; i++) {
        int pid = fork();

        if (pid > 0) {
            wait(NULL);
        } else if (pid == 0) {
            printf("Child%d's pid: %d\n", i, getpid());
            _exit(0);
        } else {
            printf("Fork Failed for Child%d\n", i);
        }
    }
}

File: Java/Sec-A/assignment1/question10.java
/*  Q10. Write a java program that will print the following patterns
             *
           * * *
         * * * * *
       * * * * * * *
     * * * * * * * * *
 */

import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = sc.nextInt();

        for (int i = 1; i <= num; i++) {
            for (int j = i + 1; j <= num; j++) {
                System.out.print("  ");
            }
            for (int j = 1; j <= (i * 2) - 1; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }
}
